amip_interp/include/amip_interp.inc

```diff
@@ -519,8 +180,8 @@ else
 
     mobs_sst = 1440;  nobs_sst = 720
 
-    call set_sst_grid_edges_daily(mobs_sst, nobs_sst)
-    call horiz_interp_new ( Interp%Hintrp2, lon_bnd, lat_bnd, &
+    call SET_SST_GRID_EDGES_DAILY_ (mobs_sst, nobs_sst)
+    call horiz_interp_new ( Interp%Hintrp2, LON_BND_, LAT_BND_, &
```

Comment:
Curious as to how this work.  Following how lon_bnd and lat_bnd are set in amip_interp_init , they're always going to be in r8_kind.  Wondering what happens if lon_model and lat_model are r4_kind...

Reply:
In that case, the module should be initialized with `amip_interp_init_r4` rather than `amip_interp_init` (which is just an alias for `amip_interp_init_r8`), so that `LON_BND_` and `LAT_BND_` refer to the `r4_kind` versions.

Reply:
I think we're trying to minimize the number of changes required on the components side.  It would be nice if the users can just call amip_interp_init instead of making changes to their code to call amip_interp_init_r4

Reply:
It's a similar problem to `data_override`, where the initialization routine allocates real data but doesn't take any real arguments. We could take a similar approach here, e.g.,`amip_interp_init(mode=r4_kind)` or `amip_interp_init(mode=r8_kind)`, but I don't think we can avoid user code having to explicitly specify the kind unless we make `tempamip`, `lon_bnd`, and `lat_bnd` all `r8_kind`. Or perhaps make `amip_interp_init` a no-op and have `amip_interp_new` do the actual initialization.

Reply:
Would it bad to make those arrays r8_kind?  It seems like that would make things simpler..

Reply:
We could, although I'm not sure what the performance implications (if any) would be.